dali/operators/reader/loader/numpy_loader.cc

```diff
@@ -38,6 +37,75 @@ TypeInfo TypeFromNumpyStr(const std::string &format) {
   DALI_FAIL("Unknown Numpy type string");
 }
 
+void ParseHeaderMetadata(NumpyParseTarget& target, std::string header) {
+  header.erase(std::remove_if(header.begin(), header.end(), ::isspace), header.end());
+
+  const char t1[] = "{'descr':\'";
+  size_t l1 = strlen(t1);
+  DALI_ENFORCE(strncmp(header.c_str(), t1, l1) == 0);
+  size_t pos = l1;
+  auto iter = header.find('\'', pos);
+  DALI_ENFORCE(iter != std::string::npos);
+
+  std::string typestring = header.substr(pos, iter - pos);
+  pos = iter + 1;
+
+  // < means LE, | means N/A, = means native. In all those cases, we can read
+  bool little_endian =
+    (typestring[0] == '<' || typestring[0] == '|' || typestring[0] == '=');
+  DALI_ENFORCE(little_endian,
+    "Big Endian files are not supported.");
+
+  std::string tid = typestring.substr(1);
+  // get type in a safe way
+  target.type_info = TypeFromNumpyStr(tid);
+
+  const char t2[] = "\'fortran_order\':";
+  size_t l2 = strlen(t2);
+
+  iter = header.find(t2, pos);
+  DALI_ENFORCE(iter != std::string::npos);
+  pos = iter + l2;
+
+  iter = header.find(',', pos);
+  auto fortran_order_str = header.substr(pos, iter - pos);
+  pos = iter + 1;
+  if (fortran_order_str == "True") {
+    target.fortran_order = true;
+  } else if (fortran_order_str == "False") {
+    target.fortran_order = false;
+  } else {
+    DALI_FAIL("Can not parse fortran order");
+  }
```

Comment:
Here again we crate a temporary string that's not exactly necessary and we could use strncmp and `header` with offset to `pos` 

Still we create a lot of them below, so...